src/data/general-cs-quiz.ts

```diff
@@ -1530,6 +1533,38 @@ const generalCSQuiz = [
     Explanation:
       "Ternary Search is similar to binary search but distinguishes itself through a trisection approach giving us three possibilities for the two points m1 and m2 f(m1) < f(m2), f(m1) > f(m2), f(m1) = f(m2)",
     Link: "https://en.wikipedia.org/wiki/Ternary_search"
+  },
+  {
+    Question:
+      "In RotateLeft or RotateRight function in an AVL tree, When do we update the root of the tree [we pass node as a argument to the function]",
+    Answer: "when Parent of node becomes nullptr",
+    Distractor1: "when node becomes nullptr",
+    Distractor2: "when node becomes a leaf of the tree",
+    Distractor3: "when the height balance is restored ",
+    Explanation:
+      "When the parent of the node becomes nullptr, this means the  node is the root of the tree,so when we make right or left rotation we need to update the root of the tree with appropriate child of node.",
+    Link: "https://forum.freecodecamp.org/t/algorithms-avl-trees/297458"
+  },
+  {
+    Question: "In Red-Black Tree, Every tree leaf node is ?",
+    Answer: "Black",
+    Distractor1: "Red",
+    Distractor2: "Red or Black",
+    Distractor3: "White",
+    Explanation:
+      "To maintain the black height and to ensure that the red child is not having a red node as its parent.",
+    Link: "https://forum.freecodecamp.org/t/red-black-trees/297473"
+  },
+  {
+    Question:
+      "What is a data structure that consists of nodes, where each node has data and a reference to the next node in the sequence?",
+    Answer: "Linked List",
+    Distractor1: "Vector",
+    Distractor2: "Stack",
+    Distractor3: "Heap",
+    Explanation:
+      "A Linekd List contains node and each node has data and a reference to next node so that it can have access to the data of its next node and we can get the informstion of all nodes by iterting the node until it reaches nullptr.",
+    Link: "https://forum.freecodecamp.org/t/freecodecamp-challenge-guide-create-a-linked-list-class/301628"
```

Comment:
you should always point to articles instead of solutions to freeCodeCamp challenges

```suggestion
    Link: "https://www.freecodecamp.org/news/how-linked-lists-work/"
```